lib/toml-rb/parser.rb

```diff
@@ -19,18 +20,20 @@ def initialize(content, options = {})
     # Read about the Visitor pattern
     # http://en.wikipedia.org/wiki/Visitor_pattern
     def visit_table_array(table_array)
+      @fully_defined_keys = []
       table_array_key = table_array.full_key
       @visited_keys.reject! { |k| k.start_with? table_array_key }
 
       @current = table_array.navigate_keys @hash, @symbolize_keys
     end
 
-    def visit_keygroup(keygroup)
-      @current = keygroup.navigate_keys @hash, @visited_keys, @symbolize_keys
+    def visit_table(table)
+      @fully_defined_keys = []
```

Comment:
Who uses this variable?

Reply:
@fully_defined_keys is used as a memory for inline table.
Inline tables can not be used to add keys or sub-tables to an already-defined table.
See https://github.com/toml-lang/toml#inline-table

So, I add this variable, and tests for this problem.
[test/examples/invalid/inline-table-duplicate-key.toml](https://github.com/emancu/toml-rb/blob/d86c537a2200261f4274f4085efad5dfe3ce34ed/test/examples/invalid/inline-table-duplicate-key.toml)
[test/examples/invalid/inline-table-duplicate-key2.toml](https://github.com/emancu/toml-rb/blob/d86c537a2200261f4274f4085efad5dfe3ce34ed/test/examples/invalid/inline-table-duplicate-key2.toml)